gap/classical.gi

```diff
@@ -581,13 +585,14 @@ IsPrimitivePrimeDivisor := function( b, a, p )
 
     local i;
 
-    if not IsPrimeInt(p) then return false; fi;
-    if not IsInt(b) or b <= 1 then return false; fi;
-    if not IsInt(a) or a <= 1 then return false; fi;
-
-    if (b^a-1) mod p <> 0 then return false; fi;
+    if not (IsPrimeInt(p) and IsInt(a) and IsInt(b) and a > 1 and b > 1 and
+            b^a-1 mod p = 0) then
+        return false;
+    fi;
```

Comment:
Actually this is one of the few places where I kind of disagree: Separating the four checks for me makes it easier to follow the code, as they have a kind of logical grouping: First we validate, p; then b, then a. After this, we start checking the actual conditions we are supposed to check.

In an unrelated matter, I think that if `IsPrimeInt(p)` is false, then we actually should call `Error`, because that indicates a usage error, and should not be silently swallowed. And I am tempted to say the say should hold for the initial checks of a and b.

Reply:
I reverted this. In my opinion, there's a lot about `classical.gi` that could/should be cleaned up (I already started to have a go at this in a separate branch). It would probably be better to make changes of the kind I made here in a more targeted, coherent manner.